### PR TITLE
Stable file fix

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: examples/stable_memory/test/tests.ts [hunk 1]
```diff
@@ -15,9 +15,10 @@ export function getTests(
             name: 'stable size',
             test: async () => {
                 const result = await stableMemoryCanister.stableSize();
+                console.log(result);
```
Comment (lastmjs) [inline examples/stable_memory/test/tests.ts:18]: Remove log

File: src/compiler/file_uploader/upload_file.ts [hunk 2]
```diff
@@ -12,6 +13,10 @@ export async function uploadFile(
     const uploadStartTime = process.hrtime.bigint();
     const fileSize = (await stat(srcPath)).size;
     const file = await open(srcPath, 'r');
+    if (!(await shouldBeUploaded(srcPath, destPath, actor))) {
```
Comment (lastmjs) [inline src/compiler/file_uploader/upload_file.ts:16]: Make this first

File: src/compiler/file_uploader/upload_file.ts [hunk 3]
```diff
@@ -12,6 +13,10 @@ export async function uploadFile(
...
+        file.close();
```
Comment (lastmjs) [inline src/compiler/file_uploader/upload_file.ts:17]: Remove file.close()

File: src/compiler/file_uploader/upload_file.ts [hunk 4]
```diff
@@ -69,3 +74,16 @@ function calculatePercentComplete(
...
 }
+
+async function shouldBeUploaded(
+    srcPath: string,
+    destPath: string,
+    actor: UploaderActor
+): Promise<boolean> {
+    const localHash = (await hashFile(srcPath)).toString('hex');
+    const canisterHash = await actor.get_file_hash(destPath);
+    if (canisterHash.length === 0) {
```
Comment (lastmjs) [inline src/compiler/file_uploader/upload_file.ts:85]: Rename canisterHash to canisterHashOption

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/stable_memory/test/tests.ts (hunk 1 above)
Comment (bdemann) [reply examples/stable_memory/test/tests.ts:18]: done

File: src/compiler/file_uploader/upload_file.ts (hunk 2 above)
Comment (bdemann) [reply src/compiler/file_uploader/upload_file.ts:16]: done

File: src/compiler/file_uploader/upload_file.ts (hunk 3 above)
Comment (bdemann) [reply src/compiler/file_uploader/upload_file.ts:17]: done

File: src/compiler/file_uploader/upload_file.ts (hunk 4 above)
Comment (bdemann) [reply src/compiler/file_uploader/upload_file.ts:85]: done

**Review by lastmjs — APPROVED** [review]

**Pull request merged**